docs/java/signals.md

```diff
@@ -3,3 +3,61 @@ id: signals
 title: Signals in Java
 sidebar_label: Signals
 ---
+
+Workflows can listen to external events and save the state accordingly.
```

Comment:
should we have something in samples-java to show signals @tsurdilo? https://github.com/temporalio/samples-java